src/components/timeline/timeline.tsx

```diff
@@ -8,26 +8,58 @@ import { Slider } from "@mui/material";
 import css from "./timeline.scss";
 
 const TICK_COUNT = 16;
+const LOADING_DELAY = 100; // ms
 
 export const Timeline: React.FC = observer(function WrappedComponent() {
-  const { simulation } = useStores();
+  const { simulation, snapshotsManager } = useStores();
+  const timeoutId = useRef(0);
   const [val, setVal] = useState(simulation.timeInYears);
   const [disabled, setDisabled] = useState(true);
+  const [timeProgress, setTimeProgress] = useState("0%");
 
   const tickDiff = simulation.config.simulationEndYear / TICK_COUNT;
   const ticks = Array.from({ length: TICK_COUNT + 1 }, (_, i) => i * tickDiff);
-  const timeProgress = `${simulation.timeInYears / simulation.config.simulationEndYear * 100}%`;
   const marks = ticks.map((tick) => ({ value: tick, label: tick }));
   useEffect(() => {
     setDisabled(!simulation.simulationStarted || (simulation.simulationRunning && !simulation.simulationEnded));
   }, [simulation.simulationStarted, simulation.simulationRunning, simulation.simulationEnded]);
 
+    // if user has scrubbed back on the timeline, and then starts the sim again
+    // we need to restore the last snapshot and move the timeline scrubber to the max year
+    useLayoutEffect(() => {
+    if (simulation.simulationRunning && snapshotsManager.maxYear > val) {
+      snapshotsManager.restoreLastSnapshot();
+      simulation.start(); //when we restoreLastSnapshot, we need to start the simulation again
+    }
+  },[simulation, simulation.simulationRunning, snapshotsManager, val]);
+
+  //we use the second useEffect to update the timeline scrubber when the simulation is running
+  // progress bar and regrowth of vegetation
   useLayoutEffect(() => {
-    setVal(simulation.timeInYears);
-  }, [simulation.timeInYears]);
+    if (simulation.simulationRunning) {
+      if (snapshotsManager.maxYear > val) {
+        setVal(snapshotsManager.maxYear);
+        simulation.setTimeInYears(snapshotsManager.maxYear);
+        setTimeProgress(`${snapshotsManager.maxYear / simulation.config.simulationEndYear * 100}%`);
+      }
+    } else if (!simulation.simulationStarted) { // if the simulation is reloaded, reset the timeline
+      setTimeProgress("0%");
+      setVal(0);
+      simulation.setTimeInYears(0);
+    } else {
+      setVal(simulation.timeInYears);
+    }
+// adding val to the dependencies array causes the thumbnail to be less responsive
```

Comment:
> // adding val to the dependencies array causes the thumbnail to be less responsive

But not adding it means that it's not working as you might think it works. The `val` value will be outdated in this block in some cases. I know it might sound like nitpicking, but I think it'd be worth to investigate what actually happens here.

Reply:
Good nitpicking! I finally realized why adding `val` as a dependency was causing the thumb to be sluggish. I was calling `setVal` twice, once in `handleSliderChange`, and another in the `else` part of the `useLayoutEffect` logic.